tests/test_ilastik/test_applets/batchProcessing/test_batchProcessingGui.py

```diff
@@ -207,10 +207,10 @@ def assert_raises(exception_type):
     "secondary_files,expectation",
     [
         (tuple(), does_not_raise),
-        (("sec1",), assert_raises(BatchProcessingDataConstraintException)),
+        (("sec1",), does_not_raise),
```

Comment:
could you please comment on this?

Reply:
So for the switch from `assert_raises` to `does_not_raise`: It is perfectly legitimate to run a batch where some of the iterations of the batch do not have a matching second role; For example, in pixel classification, it is totally fine to have the first iteration of the batch to have `Raw Data` (role 1) and `Prediction Mask` (role 2), and then have the second iteration of the batch only have `Raw Data`, since `Prediction Mask` is an optional input, anyway.

Of course, what role is or isn't optional is determined by the workflow (not the Gui), and checked by the `DataSelectionApplet`; Ideally we should instantiate (rather than mock) a `DataSelectionApplet` and use that for testing, but at this time it is so entangled with everything else that it is really hard to do that.

Reply:
okay I guess I understand this now. I still think that the functionality of the _batch processing applet_ to raise an exception whenever it determines that roles are not matching. Hence the mocking. With that code path removed I'm not sure this test makes any sense at all.